dev/RadioButtons/RadioButtons.cpp

```diff
@@ -7,136 +7,634 @@
+{
+    if (auto const element = args.Element())
+    {
+        if (auto const childHandlers = element.GetValue(s_childHandlersProperty).try_as<ChildHandlers>())
+        {
+            childHandlers->checkedRevoker.revoke();
+            childHandlers->uncheckedRevoker.revoke();
+        }
+    }
 }
 
-void RadioButtons::OnListViewKeyDown(const winrt::IInspectable& sender, const winrt::KeyRoutedEventArgs& args)
+void RadioButtons::OnRepeaterElementIndexChanged(const winrt::ItemsRepeater&, const winrt::ItemsRepeaterElementIndexChangedEventArgs& args)
 {
-    if (args.Key() == winrt::VirtualKey::Control)
+    if (auto const element = args.Element())
     {
-        m_isControlDown = true;
+        element.SetValue(winrt::AutomationProperties::PositionInSetProperty(), box_value(args.NewIndex() + 1));
     }
-    else if (args.Key() == winrt::VirtualKey::Down)
+}
+
+
+void RadioButtons::OnRepeaterCollectionChanged(const winrt::IInspectable&, const winrt::IInspectable&)
+{
+    if (auto const repeater = m_repeater.get())
     {
-        args.Handled(MoveSelection(1));
+        if (auto const itemSourceView = repeater.ItemsSourceView())
+        {
+            auto const count = itemSourceView.Count();
+            for (auto index = 0; index < count; index++)
+            {
+                if (auto const radioButton = repeater.TryGetElement(index))
+                {
+                    radioButton.SetValue(winrt::AutomationProperties::SizeOfSetProperty(), box_value(count));
+                }
+            }
+        }
     }
-    else if (args.Key() == winrt::VirtualKey::Up)
+}
+
+void RadioButtons::OnSelectionChanged(const winrt::IInspectable&, const winrt::SelectionModelSelectionChangedEventArgs& args)
+{
+    auto const currentIndex = m_selectionModel.SelectedIndex();
+    auto const repeater = m_repeater.get();
+
+    if (repeater)
     {
-        args.Handled(MoveSelection(-1));
+        if (auto const previousIndices = args.PreviousSelectedIndices())
+        {
+            for (auto const previousIndex : previousIndices)
+            {
+                if (previousIndex)
+                {
+                    if (!currentIndex || previousIndex.CompareTo(currentIndex) != 0)
+                    {
+                        if (previousIndex.GetSize() == 1)
+                        {
+                            if (auto const item = repeater.TryGetElement(previousIndex.GetAt(0)))
+                            {
+                                if (auto const itemAsToggleButton = item.try_as<winrt::ToggleButton>())
+                                {
+                                    itemAsToggleButton.IsChecked(false);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    if (currentIndex && currentIndex.GetSize() == 1)
```

Comment:
do we both get null and 0 sized indices ?

Reply:
I have seen both... but if i am remembering right, only in the PreviousSelectedIndices that I added to the args. I can't say for sure if I've ever seen SelectedIndex() return a 0 sized indcies.  This also checks for index paths with size larger than 1, which would be unexpected.

---
In reply to: [347430478](https://github.com/microsoft/microsoft-ui-xaml/pull/1623#discussion_r347430478) [](ancestors = 347430478)

Reply:
No longer an issue for RadioButtons with the removal of selection model

---
In reply to: [347543249](https://github.com/microsoft/microsoft-ui-xaml/pull/1623#discussion_r347543249) [](ancestors = 347543249,347430478)